### DISCUSSION TITLE
TypeScript and Crystal libraries updated!

### DISCUSSION THREAD

**Discussion by spilth** [SongProOrg/songpro.org#34] [Announcements]
Thanks to @FiniteLooper the JavaScript library is now working, published and converted to TypeScript.

And as I'm digging into the feasibility of making a cross-platform command-line interface for SongPro, I've updated the Crystal library to support the same features as the Ruby and TypeScript libraries.

I also did a bunch of tweaking and configuring of the various repositories so they're all set up the same which main branch protection, dependabot, tags, etc.

**Comment by ChrisMBarr** [comment]
Let me know if you have any questions about doing anything in TypeScript.  From what you are saying here, a node CLI seems pretty reasonable.  That JS/TS library would basically do a large portion of what the CLI would probably need already.

**Reply by spilth** [reply]
My personal goal is a self-contained executable for Windows, Mac and Linux, without the need to install anything.

**Reply by ChrisMBarr** [reply]
oh well never mind then!  I have zero knowledge of that! haha
